### PR TITLE
Dgit UI updates

### LINKED ISSUE #4941: DGIT UI fixes and improvements
image

### PR DESCRIPTION
**Pull request by joeizang**
fixes #4941

### CODE REVIEW

File: libs/remix-ui/git/src/components/panels/tokenWarning.tsx [hunk 1]
```diff
@@ -1,12 +1,20 @@
...
   const context = React.useContext(gitPluginContext)
   return (<>
     {(context.gitHubUser && context.gitHubUser.login) ? null :
-      <li className="text-warning list-group-item d-flex justify-content-between align-items-center">
-        To use add a GitHub token to the settings.</li>
+      <span className="text-warning text-left">
+        <span>Generate and add a Git token to use this plugin. Tokens are added in </span><span className=" text-decoration-line-through messageTip" onClick={async () => {
+          await props.plugin.call('menuicons', 'select', 'settings')
+        }}>settings</span><span> of the IDE.</span>
+      </span>
```
Comment (bunsenstraat) [inline libs/remix-ui/git/src/components/panels/tokenWarning.tsx:16]: that is not true.... it's done in the git settings of git plugin. different system.

File: libs/remix-ui/git/src/components/navigation/branches.tsx [hunk 2]
```diff
@@ -20,13 +20,11 @@ export const BranchesNavigation = ({ eventKey, activePanel, callback }) => {
     <>
       <div className={'d-flex justify-content-between pt-1 ' + (activePanel === eventKey? 'bg-light': '')}>
         <span data-id='branches-panel' onClick={()=>handleClick()} role={'button'} className='nav d-flex justify-content-start align-items-center w-75'>
-          {
-            activePanel === eventKey ? <FontAwesomeIcon className='' icon={faCaretDown}></FontAwesomeIcon> : <FontAwesomeIcon className='' icon={faCaretRight}></FontAwesomeIcon>
-          }
-          <label className="pl-1 nav form-check-label">BRANCHES</label>
+          <i className={ activePanel !== eventKey ? "fa fa-caret-right" : "fa fa-caret-down" }></i>
```
Comment (bunsenstraat) [inline libs/remix-ui/git/src/components/navigation/branches.tsx:23]: is this replacing fontawesome icon? Is that necessary. seems unpractical to me to go from typed to untyped. strong chance of typing errors.

File: libs/remix-ui/git/src/components/gitui.tsx [hunk 3]
```diff
@@ -200,37 +200,38 @@ export const GitUI = (props: IGitUi) => {
...
+                      </div>
                     </Accordion.Collapse>
                     <hr></hr>
                     <RemotesNavigation eventKey="5" activePanel={activePanel} callback={setActivePanel} />
                     <Accordion.Collapse className='bg-light' eventKey="5">
-                      <>
-                        <Remotes></Remotes>
-                      </>
+                      <div className="px-2 py-2">
+                        <Remotes plugin={plugin}></Remotes>
```
Comment (bunsenstraat) [inline libs/remix-ui/git/src/components/gitui.tsx:211]: don't do this. it's not needed. the plugin is in the context if you need to call it.

File: libs/remix-ui/git/src/components/panels/tokenWarning.tsx [hunk 4]
```diff
@@ -1,12 +1,20 @@
 import { gitPluginContext } from "../gitui"
-import React, { useEffect, useState } from "react";
-export const TokenWarning = () => {
+import React, { useEffect, useState } from "react"
+
+export interface TokenWarningProps {
+  plugin: any
```
Comment (bunsenstraat) [inline libs/remix-ui/git/src/components/panels/tokenWarning.tsx:5]: plugin is in the context no need to pass it around

File: libs/remix-ui/git/src/components/panels/tokenWarning.tsx (hunk 1 above)
Comment (joeizang) [reply libs/remix-ui/git/src/components/panels/tokenWarning.tsx:16]: Ok. Will revert. Thanks @bunsenstraat

File: libs/remix-ui/git/src/components/gitui.tsx (hunk 3 above)
Comment (joeizang) [reply libs/remix-ui/git/src/components/gitui.tsx:211]: done

File: libs/remix-ui/git/src/components/panels/tokenWarning.tsx (hunk 4 above)
Comment (joeizang) [reply libs/remix-ui/git/src/components/panels/tokenWarning.tsx:5]: 100%. You are right.

File: libs/remix-ui/git/src/components/navigation/branches.tsx (hunk 2 above)
Comment (joeizang) [reply libs/remix-ui/git/src/components/navigation/branches.tsx:23]: Reverting

File: libs/remix-ui/git/src/components/panels/tokenWarning.tsx (hunk 1 above)
Comment (bunsenstraat) [reply libs/remix-ui/git/src/components/panels/tokenWarning.tsx:11]: @joeizang please do yes...

**Review by Aniket-Engg — APPROVED** [review]
Approving as suggested by @bunsenstraat

**Pull request merged**